src/odemis/acq/stream/_projection.py

```diff
@@ -218,6 +223,437 @@ def _updateImage(self):
+                                     "azimuthal projection in reduced precision.",
+                                     data_raw_bg_corr[key].shape)
+                        y, x = data_raw_bg_corr[key].shape
+                        if y > x:
+                            small_shape = 1024, int(round(1024 * x / y))
+                        else:
+                            small_shape = int(round(1024 * y / x)), 1024
+                        # resize
+                        data_raw_bg_corr[key] = img.rescale_hq(data_raw_bg_corr[key], small_shape)
+
+                # convert data into rectangular format (theta-phi-representation)
+                # Check if rectangular converted representation already was calculated for requested ebeam pos.
+                # Note: This calc is very time consuming. Takes about 3.6 sec for one ebeam pos (conversion of 6 images)
+                # and tested on an image of size (256, 1024)
+                try:
+                    polarimetry_dict_raw[ebeam_pos]
+
+                except KeyError:
+
+                    # Get the center wavelength of the filter used (no filter aka "pass-through" use fallback)
+                    # Does not matter from which of the 6 images as they all must have been recorded with the same
+                    # filter -> reuse key  # TODO dont like this solution, is there a better one?
+                    fallback_wl = 650e-9
+                    band = data_raw_bg_corr[key].metadata.get(model.MD_OUT_WL, fallback_wl)
+                    if isinstance(band, tuple):  # wl is usually tuple of min/max value
+                        wl = sum(band) / len(band)
+                    else:  # handles if band is str TODO if "str", support center wavelength based on color
+                        wl = fallback_wl
+
+                    # Warning: allocates lot of memory, which will not be free'd until
+                    # the current thread is terminated.
+
+                    # TODO allow variable input size? Calc based on raw data?
+                    size = (180, 720)  # (2*90, 2*360) defines the resolution of the displayed image
+
+                    for key in data_raw_bg_corr.keys():
+                        data_raw_bg_corr[key] = angleres.AngleResolved2Rectangular(data_raw_bg_corr[key], size)
+                    polarimetry_dict_raw[ebeam_pos] = data_raw_bg_corr  # dict containing the bg corrected raw data
+
+                    # Calculate the polarimetry results:
+                    # Dictionary keys are (ebeam pos x, ebeam pos y, optional: polarimetry key).
+                    # Note: Takes about 0.25 sec to calc all polarimetry results for one ebeam pos
+                    # and tested on an image of size (256, 1024)
+                    polarimetry_dict_raw.update(
+                        arpolvis.calcPolarimetry(polarimetry_dict_raw, ebeam_pos, size, wl))
+
+                # Convert the data back to polar representation for GUI display.
```

Comment:
It's not really converting *back*, is it? IIRC, it's just raw -> rectangular -> polar.